app/models/postcode_local_restriction_search.rb

```diff
@@ -0,0 +1,53 @@
+class PostcodeLocalRestrictionSearch
```

Comment:
I can appreciate the concern here, I wasn't sure model was the most natural place for this. It seemed to be ok for this app since we don't already have a world of ActiveRecord models to make things complex. 

It doesn't seem to represent a service to me, which I know is a wooly concept in Rails land. As far as I'm aware, the most common definition of a service object is a pattern to implement doing a single business transaction. Thus services are often represent by a verb (`TweetCreator`) of what they do and, given they only do that one transaction, they should only have a single method as their interface. This class however seems to be modelling something stateful, a user created a search - it has properties such as their input and then data derived from that state.

If it wasn't to be a model I'd consider it to be something that could be represented as a [form object](https://thoughtbot.com/blog/activemodel-form-objects) - though it doesn't seem a common enough concern for this app that we need to introduce that abstraction.